src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -0,0 +1,640 @@
+                _cleanup_free_ char *e_only_show_in = NULL;
+                _cleanup_free_ char *e_not_show_in = NULL;
+
+                only_show_in = strv_join(service->only_show_in, ":");
+                not_show_in = strv_join(service->not_show_in, ":");
+                if (!only_show_in || !not_show_in)
+                        return log_oom();
+
+                e_only_show_in = cescape(only_show_in);
+                e_not_show_in = cescape(not_show_in);
+                if (!e_only_show_in || !e_not_show_in)
+                        return log_oom();
+
+                /* Just assume the values are reasonably sane */
+                fprintf(f,
+                        "ExecCondition=" ROOTLIBEXECDIR "/systemd-xdg-autostart-condition \"%s\" \"%s\"\n",
+                        e_only_show_in,
+                        e_not_show_in);
+        }
+
+        /* Generate an ExecCondition for GNOME autostart condition */
+        if (!isempty(service->autostart_condition)) {
+                _cleanup_free_ char *gnome_autostart_condition_path = NULL;
+                _cleanup_free_ char *e_autostart_condition = NULL;
+
+                if (find_binary("gnome-systemd-autostart-condition", &gnome_autostart_condition_path) < 0) {
+                        e_autostart_condition = cescape(service->autostart_condition);
+                        if (!e_autostart_condition)
+                                return log_oom();
+
+                         fprintf(f,
+                                 "ExecCondition=%s --condition \"%s\"\n",
+                                 gnome_autostart_condition_path,
+                                 e_autostart_condition);
+                 } else {
+                         log_info_errno(r, "gnome-systemd-autostart-condition not found: %m");
+                         fprintf(f, "# ExecCondition for GNOME skipped due to missing binary\n");
+                }
+        }
+
+        /* Generate an ExecCondition for KDE autostart condition */
+        if (!isempty(service->kde_autostart_condition)) {
+                _cleanup_free_ char *kde_autostart_condition_path = NULL;
+                _cleanup_free_ char *e_kde_autostart_condition = NULL;
+
+                if (find_binary("kde-systemd-start-condition", &kde_autostart_condition_path) < 0) {
+                        e_kde_autostart_condition = cescape(service->kde_autostart_condition);
+                        if (!e_kde_autostart_condition)
+                                return log_oom();
+
+                        fprintf(f,
+                                "ExecCondition=%s \"%s\"\n",
+                                kde_autostart_condition_path,
+                                e_kde_autostart_condition);
+                } else {
+                        log_info_errno(r, "kde-systemd-start-condition not found: %m");
+                        fprintf(f, "# ExecCondition for KDE skipped due to missing binary\n");
+                }
```

Comment:
hmm, the gnome and kde lines are pretty similar, maybe turn this into a helper function that is just called with slightly different args?